lib/backends/smartos/tasks/recovery_config.js

```diff
@@ -0,0 +1,282 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+/*
+ * Copyright 2019 Joyent, Inc.
+ */
+
+/*
+ * This task stage/activate the given recovery configuration into
+ * a CN using EDAR (i.e. zpool encryption must be true)
+ */
+
+
+var child_process = require('child_process');
+var crypto = require('crypto');
+var execFile = child_process.execFile;
+var fs = require('fs');
+var util = require('util');
+
+var async = require('async');
```

Comment:
Maybe use vasync instead?

Reply:
Yeah. I've also moved to just use `pipeline` instead of `waterfall`. Not taking advantage of it at all.